### DISCUSSION TITLE
COLOR_A support?

### DISCUSSION THREAD

**Discussion by jwestes** [RudolphRiedel/FT800-FT813#70] [Q&A] [answered]
I want to make an overlay that will have a semi-transparent background.  I figured out how to do it with EVE Screen Editor, but this library does not appear to support COLOR_A.  Am I just missing something?

This is what I am trying to do:
    EVE_cmd_dl_burst(LINE_WIDTH(1 * 16));
    EVE_cmd_dl_burst(DL_BEGIN | EVE_RECTS);
    EVE_cmd_dl_burst(DL_COLOR_RGB | BLACK);
    EVE_cmd_dl_burst(DL_COLOR_A | 0xEE); //I cannot find how to set the alpha channel.
    EVE_cmd_dl_burst(VERTEX2F(0, 0));
    EVE_cmd_dl_burst(VERTEX2F(800, 480));
    EVE_cmd_dl_burst(DL_END);

**Comment by RudolphRiedel** [comment, marked as answer]
It is defined as macro, use COLOR_A(value) - one of these things that I never actually used and that is one of the original macros.
EVE_cmd_dl_burst(COLOR_A(0xEE));

Somehow this is all inconsistent by now.

Looking at the current EVE Screen Editor I should add a couple more functions.
I already added EVE_color_rgb().

void EVE_color_a(uint8 alpha);

Plus a couple of others like clear, color_mask, line_width, point_size and so on, everything that has not CMD_ in the name in ESE and still looks like a function while it originally was implemented as a macro to provide a 32 bit value for a cmd_dl() call.

Well, plus adding a couple more DL_ defines would not hurt either.
The macros provide range limits for the parameters though.

**Comment by jwestes** [comment]
That did it.  Thanks!
